AppCenterDataStorage/AppCenterDataStorageTests/MSDataSourceErrorTests.m

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#import "AppCenter.h"
+#import "MSDataSourceError.h"
+#import "MSDataStoreErrors.h"
+#import "MSTestFrameworks.h"
+
+@interface MSDataSourceErrorTests : XCTestCase
+
+@end
+
+@implementation MSDataSourceErrorTests
+
+- (void)testInitWithErrorCallsParsingMethod {
+
+  // If
+  NSInteger expectedErrorCode = 500;
```

Comment:
There's an enum in `MSConstants+Internal.h` with status codes - this one would be `MSHTTPCodesNo500InternalServerError`

Reply:
As far as I know, we have a separate [enum with storage errors](https://github.com/Microsoft/AppCenter-SDK-Apple/blob/develop/AppCenterDataStorage/AppCenterDataStorage/MSDataStoreErrors.h#L16) so I used it instead of `MSConstants+Internal.h`. Please let me know if it looks good to you.